File: graphr/src/graphr/graph/GraphData.java
package graphr.graph;

import java.io.Serializable;

public interface GraphData extends Serializable {

	/**
	 * Part of the visitor design pattern -all graph data must implement accept method
	 * @param visitor Reference to a particular visitor
	 * @return Object that is serializable
	 */
	Serializable accept(GraphDataVisitor visitor);

}

File: graphrs/graphrs/graph/GraphData.swift
//
//  GraphData.swift
//  graphrs
//
//  Created by Marcus Sellmann on 24.03.16.
//  Copyright © 2016 CITBDA. All rights reserved.
//

import Foundation

protocol GraphData {
    /**
     * Part of the visitor design pattern -all graph data must implement accept method
     * @param visitor Reference to a particular visitor
     * @return Object that is serializable
     */
    func accept(visitor: GraphDataVisitor) -> AnyObject
}